src/languages/c-like.js

```diff
@@ -200,6 +201,58 @@ export default function(hljs) {
     ]
   };
 
+  var TEMPLATE_USE = {
+    begin: /</, end: />/,
+    keywords: CPP_KEYWORDS,
+    contains: EXPRESSION_CONTAINS.concat([
+      // Match left-shift to prevent it from creating a nested `TEMPLATE_USE`.
+      { begin: /<</ },
+      'self'
+    ])
+  };
+
+  var CLASS_DECLARATION = {
+    className: 'class',
+    beginKeywords: 'class struct union', end: /[{;:]/,
+    contains: [
+      { beginKeywords: "final" },
+      TEMPLATE_USE,
+      TITLE_MODE
+    ]
+  };
+
+  var TEMPLATE_DECLARATION = {
+    beginKeywords: 'template',
+    // Add additional stops to stop runaway template declaration, e.g.
+    // `template <bool b = 1 < 2> void f();`
+    end: /[>;{]/,
+    contains: [
+      {
+        begin: /</, end: />/,
+        endsWithParent: true,
+        endsParent: true,
+        keywords: CPP_KEYWORDS,
+        contains: EXPRESSION_CONTAINS.concat([
+          // Match left-shift to prevent it from creating a nested `TEMPLATE_USE`.
+          { begin: /<</ },
+          TEMPLATE_USE,
+          {
+            begin: /\(/, end: /\)/,
+            keywords: CPP_KEYWORDS,
+            contains: [
+              // Match left-shift to prevent it from creating a nested `TEMPLATE_USE`.
+              { begin: /<</ },
+              TEMPLATE_USE
+            ]
+          },
+        ]),
+        relevance: 10
+      },
+      // FIXME: Set relevance of class declarations at this point to 10!
```

Comment:
Can this be done without copying the definition of `CLASS_DECLARATION`? (It’s probably trivial but despite appearances to the contrary I don’t actually know JavaScript.)

Reply:
`inherit`